src/accounts/accounts.controller.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  Controller,
+  Post,
+  Body,
+  HttpException,
+  HttpStatus,
+} from '@nestjs/common';
+import { Accounts } from './accounts.model';
+import { AccountsService } from './accounts.service';
+import { AuthService } from './auth.service';
+import { Sequelize } from 'sequelize-typescript';
+
+@Controller('accounts')
+export class AccountsController {
+  constructor(
+    private readonly accountsService: AccountsService,
+    private readonly authService: AuthService,
+    private readonly sequelize: Sequelize,
+  ) {}
+
+  @Post()
+  async createAccount(
+    @Body() accountData: Accounts,
+  ): Promise<{ data: Accounts; authToken: string }> {
+    const transaction = await this.sequelize.transaction();
```

Comment:
suggest using sequelize.transaction((transaction)=>{}) approach (managed transactions)

doing this will automatically rollback the transaction in case of errors.